bld/src/libiotsdk/CMakeLists.txt

```diff
@@ -6,6 +6,22 @@
 # a subset and replace other parts in the public shared sdk, e.g. logging.
 #
 
+if (CMAKE_VERSION VERSION_LESS "3.1")
+	if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
+          # Adding --std=c99 to CMAKE_C_FLAGS breaks compilation of several subprojects!
+          # e.g libwebsockets but ${_SHARED_ROOT} needs it!
+	  set (CMAKE_C_FLAGS "--std=c99 ${CMAKE_C_FLAGS}")
+          set (CMAKE_CXX_FLAGS "--std=c++11 ${CMAKE_CXX_FLAGS}")
+          # Macro _POSIX_C_SOURCE is needed by ${_SHARED_ROOT}
+          set (CMAKE_C_FLAGS "-D_POSIX_C_SOURCE=200112L ${CMAKE_C_FLAGS}")
+	endif()
+else()
+	set (CMAKE_C_STANDARD 99)
+    set (CMAKE_CXX_STANDARD 11)
+          # Macro _POSIX_C_SOURCE is needed by ${_SHARED_ROOT}
+          set (CMAKE_C_FLAGS "-D_POSIX_C_SOURCE=200112L ${CMAKE_C_FLAGS}")
```

Comment:
It is already in the shared project:
https://github.com/Azure/azure-c-shared-utility/blob/master/CMakeLists.txt#L177
But the CMakeList.txt of the shared project is not used.
So I get the following output without this macro set here:

make[2]: Entering directory '/home/klj/iot-gateway-proxy/build/cmake/Release'
/usr/bin/cmake -E cmake_progress_report /home/klj/iot-gateway-proxy/build/cmake/Release/CMakeFiles 
[  0%] Building C object bld/src/libiotsdk/CMakeFiles/libiotsdk.dir/__/__/__/deps/shared/adapters/condition_pthreads.c.o
cd /home/klj/iot-gateway-proxy/build/cmake/Release/bld/src/libiotsdk && /usr/bin/cc  -DNO_ZLOG -DUSE_OPENSSL=1 --std=c99  -O3 -DNDEBUG -fPIC -I/home/klj/iot-gateway-proxy/deps/shared/inc    -o CMakeFiles/libiotsdk.dir/__/__/__/deps/shared/adapters/condition_pthreads.c.o   -c /home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c:22:1: error: unknown type name ‘clockid_t’
 clockid_t time_basis = -1;
 ^
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c: In function ‘set_time_basis’:
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c:40:5: warning: implicit declaration of function ‘pthread_condattr_getclock’ [-Wimplicit-function-declaration]
     pthread_condattr_getclock(&cattr, &time_basis);
     ^
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c: In function ‘create_cond’:
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c:56:9: warning: implicit declaration of function ‘pthread_condattr_setclock’ [-Wimplicit-function-declaration]
         pthread_condattr_setclock(&cattr, time_basis);
         ^
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c: At top level:
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c:65:24: warning: ‘struct timespec’ declared inside parameter list
 int get_time_ns(struct timespec* ts)
                        ^
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c: In function ‘get_time_ns’:
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c:85:5: warning: implicit declaration of function ‘clock_gettime’ [-Wimplicit-function-declaration]
     err = clock_gettime(time_basis, ts);
     ^
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c: In function ‘Condition_Wait’:
/home/klj/iot-gateway-proxy/deps/shared/adapters/condition_pthreads.c:140:29: error: storage size of ‘tm’ isn’t known
             struct timespec tm;
                             ^
bld/src/libiotsdk/CMakeFiles/libiotsdk.dir/build.make:721: recipe for target 'bld/src/libiotsdk/CMakeFiles/libiotsdk.dir/__/__/__/deps/shared/adapters/condition_pthreads.c.o' failed